ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -27,8 +27,261 @@
+    }
+
+    SECTION("Simple put test", "[gcp-connection]")
+    {
+
+        // Upload a file that does not exist locally - should fail.
+        REQUIRE(conn.put_object(non_existant_object_name, non_existant_file_name) == ENOENT);
+
+        // Check number of files with the given prefix that are currently in the bucket.
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 0));
+
+        // Upload a test file.
+        REQUIRE(conn.put_object(object_name, "./" + file_name) == 0);
+
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 1));
+
+        // Check the bucket contains the uploaded file.
+        bool exists = check_exist(client, test_bucket_name, test_bucket_prefix, object_name);
+        REQUIRE(exists);
+
+        // Delete the uploaded file.
+        client.DeleteObject(test_bucket_name, test_bucket_prefix + object_name);
+
+        exists = check_exist(client, test_bucket_name, test_bucket_prefix, object_name);
+        REQUIRE_FALSE(exists);
+    }
+
+    SECTION("Simple delete test", "[gcp-connection]")
+    {
+
+        // Delete a file that does not exist in the bucket - should fail.
+        REQUIRE(conn.delete_object(non_existant_object_name) == ENOENT);
+
+        // Upload a test file.
+        auto metadata = upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name);
+        REQUIRE(metadata.ok());
+
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 1));
+
+        // Delete the uploaded file.
+        REQUIRE(conn.delete_object(object_name) == 0);
+
+        // Check that the file has been deleted.
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 0));
+
+        bool exists = check_exist(client, test_bucket_name, test_bucket_prefix, object_name);
+        REQUIRE_FALSE(exists);
+        REQUIRE_FALSE(check_exist(client, test_bucket_name, test_bucket_prefix, object_name));
+    }
+
+    SECTION("Simple object exists test", "[gcp-connection]")
+    {
+        bool exists;
+        size_t size;
+
+        REQUIRE(conn.object_exists(object_name, exists, size) == 0);
+        REQUIRE(exists == false);
+        REQUIRE(size == 0);
+
+        // Upload a test file.
+        auto metadata = upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name);
+        REQUIRE(metadata.status().ok());
+
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 1));
+
+        // Check the bucket contains the uploaded file.
+        REQUIRE(conn.object_exists(object_name, exists, size) == 0);
+        REQUIRE(exists);
+        REQUIRE(size != 0);
+
+        // Delete the uploaded file.
+        auto dl_metadata = client.DeleteObject(test_bucket_name, test_bucket_prefix + object_name);
+        REQUIRE(dl_metadata.ok());
+
+        // Check that the file has been deleted.
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 0));
```

Comment:
nit: This works but an explicit object exists check would be more accurate here.

Reply:
addressed here: [8039651](https://github.com/wiredtiger/wiredtiger/pull/8749/commits/8039651071462d3bd21d784fdb335f39875e742f)